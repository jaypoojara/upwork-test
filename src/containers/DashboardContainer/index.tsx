import {
  Button,
  Container,
  Typography
} from '@mui/material';
import { ApexOptions } from 'apexcharts';
import Chart from 'react-apexcharts';
import React from 'react';
import { useIntl } from 'react-intl';
import {
  useDispatch,
  useSelector
} from 'react-redux';
import { appContainerCreators } from '../../AppReducer';
import { selectUsername } from '../../AppSelector';
import Colors from '../../theme/colors';
import data from '../../utils/data.json';


const Styles = {
  container: {
    height: '100vh',
    width: '100vw',
    alignItems: 'center',
    justifyContent: 'center',
    display: 'flex',
    flexDirection: 'column'
  }
}

export default function DashboardContainer() {
  const username = useSelector(selectUsername);
  const intl = useIntl();
  const chartData: ApexOptions = {
      chart: {
        id: 'basic-bar'
      },
      xaxis: {
        categories: data.map((data: any) => data.date)
      },
      colors: [Colors.purple],
      series: [
        {
          name: 'Data Series',
          data: data.map((data: any) => data.value)
        }
      ]
    };

  const dispatch = useDispatch();

  const handleSignOut = () => {
    dispatch(appContainerCreators.requestLogout())
  }

  return (
    <Container component="main" sx={Styles.container}>
      <Typography component='h1' variant='h5'>
        {username}
      </Typography>
      <Chart
        options={chartData}
        series={chartData.series}
        type={'line'}
        width={800}
      />
      <Button
        data-testid={'logout-button'}
        type={'button'}
        size={'medium'}
        variant="contained"
        sx={{ mt: 3, mb: 2, width: '25%' }}
        onClick={handleSignOut}
      >
        {intl.formatMessage({ id: 'sign_out' })}
      </Button>
    </Container>
  );
}
