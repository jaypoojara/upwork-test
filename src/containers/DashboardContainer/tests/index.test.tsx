import React from 'react';
import {
  renderProvider,
} from '../../../utils/testUtils';
import DashboardContainer from '../index';

jest.mock('react-apexcharts', () => {
  return {
    __esModule: true,
    default: () => {
      return <div />
    },
  }
})


describe('<DashboardContainer /> tests', () => {
  it('should render and match the snapshot', () => {
    const {baseElement} = renderProvider(<DashboardContainer />)
    expect(baseElement).toMatchSnapshot();
  });
});
