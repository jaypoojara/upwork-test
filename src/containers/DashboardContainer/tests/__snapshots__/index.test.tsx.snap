// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardContainer /> tests should render and match the snapshot 1`] = `
<body>
  <div>
    <main
      class="MuiContainer-root MuiContainer-maxWidthLg css-1pokgtz-MuiContainer-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      />
      <div />
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-11a4262-MuiButtonBase-root-MuiButton-root"
        data-testid="logout-button"
        tabindex="0"
        type="button"
      >
        Sign out
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </main>
  </div>
</body>
`;
